File: Kattis/Divisors.py
import sys
N = 433
sieve = [False]*N
primes= []
for i in range(2,N):
    if not sieve[i]:
        primes.append(i)
        for j in range(2*i,N,i):
            sieve[j]=True

nPrimes = len(primes)

primeFacs = [[] for _ in range(N)]
primeFacs[0]=[0 for _ in range(nPrimes)]
primeFacs[1]=[0 for _ in range(nPrimes)]

for i in range(2,N):
    t = i
    facs = [0]*nPrimes
    for j in range(nPrimes):
        while i % primes[j] == 0:
            i/=primes[j]
            facs[j]+=1
    primeFacs[t] = facs

for line in sys.stdin:
    [n,k] = map(int, line.split())

    val = [0]*nPrimes

    for x in range(n-k+1,n+1)[::-1]:
        for i in range(nPrimes):
            val[i]+=primeFacs[x][i]

    for x in range(1,k+1):
        for i in range(nPrimes):
            val[i]-=primeFacs[x][i]

    out = 1
    for i in range(nPrimes):
        out*=(val[i]+1)

    print out

File: Kattis/Divisors.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;
constexpr int N = 433;
bool sieve[N];
vector<ll> primes;

int main(){
  for(int i = 2; i < N; ++i){
    if(!sieve[i]){
      primes.push_back(i);
      for(int j = 2*i; j < N; j+=i)
          sieve[j]=true;
    }
  }

  int nPrimes = primes.size();

  vector<ll> primeFacs[N];
  primeFacs[0].assign(N, 0);
  primeFacs[1].assign(N, 0);

  for(int i = 2; i < N; ++i){
      int t = i;
      ll p = i;
      vector<ll> facs(nPrimes, 0);
      for(int j = 0; j < nPrimes; ++j){
        while(p % primes[j] == 0){
          p /= primes[j];
          facs[j]+=1;
        }
      }
      primeFacs[t] = std::move(facs);
  }

  int n,k;
  while(scanf("%d %d", &n, &k) == 2) {
    vector<ll> val(nPrimes,0);

    for(int x = n-k+1; x < n+1; ++x)
      for(int i = 0; i < nPrimes; ++i)
        val[i]+=primeFacs[x][i];

    for(int x = 1; x < k+1; ++x)
      for(int i = 0; i < nPrimes; ++i)
        val[i]-=primeFacs[x][i];

    ll out = 1;
    for(int i = 0; i < nPrimes; ++i)
      out*=(val[i]+1);
    printf("%lld\n", out);
  }
}
